src/lib/db/api-token-store.ts

```diff
@@ -35,33 +36,44 @@ interface ITokenRow extends ITokenInsert {
     project: string;
 }
 
-const tokenRowReducer = (acc, tokenRow) => {
-    const { project, ...token } = tokenRow;
-    if (!acc[tokenRow.secret]) {
-        acc[tokenRow.secret] = {
-            secret: token.secret,
-            tokenName: token.token_name ? token.token_name : token.username,
-            type: token.type.toLowerCase(),
-            project: ALL,
-            projects: [ALL],
-            environment: token.environment ? token.environment : ALL,
-            expiresAt: token.expires_at,
-            createdAt: token.created_at,
-            alias: token.alias,
-            seenAt: token.seen_at,
-            username: token.token_name ? token.token_name : token.username,
-        };
-    }
-    const currentToken = acc[tokenRow.secret];
-    if (tokenRow.project) {
-        if (isAllProjects(currentToken.projects)) {
-            currentToken.projects = [];
+const createTokenRowReducer =
+    (allowOrphanedWildcardTokens?: boolean) => (acc, tokenRow) => {
+        const { project, ...token } = tokenRow;
+        if (!acc[tokenRow.secret]) {
+            if (
+                !allowOrphanedWildcardTokens &&
```

Comment:
Gosh this if statement is hairy. Honestly I'm not sure how to make this clearer so if you got no ideas I won't complain but if you have something clever to improve the readability I won't say no

Reply:
The only improvement I have in mind is not separating createTokenRowReducer into a separate function, so I can keep some variables in scope.

Complexity here can be cleaned up later on, if we decide to remove orphaned keys with migration.

Reply:
> Complexity here can be cleaned up later on, if we decide to remove orphaned keys with migration.

Yeah that's probably okay then. I'll be quiet, it can stay